ui/core/components/stat_weights_action.tsx

```diff
@@ -4,21 +4,110 @@ import { ref } from 'tsx-vanilla';
 
 import { IndividualSimUI } from '../individual_sim_ui.jsx';
 import { Player } from '../player.js';
-import { ProgressMetrics, StatWeightsResult, StatWeightValues } from '../proto/api.js';
+import { ProgressMetrics, StatWeightsResult, StatWeightValues } from '../proto/api';
 import { PseudoStat, Stat, UnitStats } from '../proto/common.js';
+import { SavedStatWeightSettings } from '../proto/ui';
 import { getStatName } from '../proto_utils/names.js';
 import { Stats, UnitStat } from '../proto_utils/stats.js';
 import { RequestTypes } from '../sim_signal_manager';
+import { SimUI } from '../sim_ui';
 import { EventID, TypedEvent } from '../typed_event.js';
 import { stDevToConf90 } from '../utils.js';
 import { BaseModal } from './base_modal.jsx';
 import { BooleanPicker } from './pickers/boolean_picker.js';
 import { NumberPicker } from './pickers/number_picker.js';
 import { ResultsViewer } from './results_viewer.jsx';
 import { renderSavedEPWeights } from './saved_data_managers/ep_weights';
+export class StatWeightActionSettings {
+	private readonly storageKey: string;
+	readonly changeEmitter = new TypedEvent<void>();
+
+	excludedStats: Stat[] = [];
+	excludedPseudoStats: PseudoStat[] = [];
+
+	constructor(simUI: SimUI) {
+		this.storageKey = simUI.getStorageKey('__statweight_settings__');
+		this.changeEmitter.on(() => {
+			const json = SavedStatWeightSettings.toJsonString(this.toProto());
+			window.localStorage.setItem(this.storageKey, json);
+		});
+	}
+
+	applyDefaults(eventID: EventID) {
+		this.excludedStats = [];
+		this.excludedPseudoStats = [];
+		this.changeEmitter.emit(eventID);
+	}
+
+	load(eventID: EventID) {
+		const storageValue = window.localStorage.getItem(this.storageKey);
+		if (storageValue) {
+			const { excludedStats, excludedPseudoStats } = SavedStatWeightSettings.fromJsonString(storageValue);
+			if (excludedStats?.length) this.excludedStats = excludedStats;
+			if (excludedPseudoStats?.length) this.excludedPseudoStats = excludedPseudoStats;
+			this.changeEmitter.emit(eventID);
+		}
+	}
+
+	toProto(): SavedStatWeightSettings {
+		const proto = SavedStatWeightSettings.create();
+		proto.excludedStats = this.excludedStats;
+		proto.excludedPseudoStats = this.excludedPseudoStats;
```

Comment:
Slice these ones as well to make safe copies. Additionally, you should set `proto.apiVersion` here.